src/EnergyPlus/DataOutputs.cc

```diff
@@ -185,9 +185,9 @@ bool FindItemInVariableList(EnergyPlusData &state, std::string const &KeyedValue
     for (auto it = found_variable->second.begin(); it != found_variable->second.end(); ++it) {
         if (equali(KeyedValue, it->second.key)) return true;
         if (it->second.is_simple_string) continue;
-        if ((it->second.pattern != nullptr && RE2::FullMatch(KeyedValue, *it->second.pattern)) || // match against regex as written
+        if ((it->second.pattern != nullptr && RE2::FullMatch(std::string{KeyedValue}, *it->second.pattern)) || // match against regex as written
             (it->second.case_insensitive_pattern != nullptr &&
-             RE2::FullMatch(KeyedValue, *it->second.case_insensitive_pattern)) // attempt case-insensitive regex comparison
+             RE2::FullMatch(std::string{KeyedValue}, *it->second.case_insensitive_pattern)) // attempt case-insensitive regex comparison
```

Comment:
I suppose these multiple string constructions are fine because these lines won't always be hit?

Reply:
I assume that at some point, all of these libraries will support `string_view` parameters where they currently take `string const &` or at least have `string_view` variants. FWIW RE2 specifically can take a `const char *` in the `KeyedValue` argument so we can pass `KeyedValue.data()` if we are sure that the string we are passing is null terminated (which I think we are, are we?).